Add element UO for Thermochimica coupling (#25880)
idaholab/moose: modules/chemical_reactions/doc/content/source/userobjects/ThermochimicaElementData.md

> # ThermochimicaElementData

Reviewer: Please use `!param` syntax for referencing parameters here

Author: done

!syntax description /UserObjects/ThermochimicaElementData

!alert note title=For Use with Thermochimica
This UserObject is designed for use with thermochemistry library Thermochimica.

## Description

> [`ThermochimicaElementData`](ThermochimicaElementData.md) perfroms Thermochimica calculations on elements.

Reviewer 2: ```suggestion
[`ThermochimicaElementData`](ThermochimicaElementData.md) performs Thermochimica calculations on elements.
```

Author: fixed

In this UserObject, the masses of elements included in the vector variable `elements` are input
to the Fortan 90 module Thermochimica, along with the temperature and pressure. Optionally, the

> user may disable Thermochimica calculation re-initialization by setting `reinit_requested` to `false`.

Reviewer 2: ```suggestion
user may disable Thermochimica calculation re-initialization by setting `reinit_type` to `none`.
```

Reviewer 2: The same change should also go in `ThermochimicaNodalData.md`

This may reduce memory use in the calculation, but will likely greatly increase the length of each
call to Thermochimica.

!alert note
Currently this object is designed to only work with constant monomial type variables (one DOF per element)!

Thermochimica is called by this object at every execute (please see the Thermochimica user manual
for more details), and the data required to re-initialize Thermochimica calculations is
saved/loaded if re-initialization is enabled.

If the optional variable `output_phases` is set to a non-empty list of phase names, then concentration
data corresponding to these phases will be output. These phase names must exactly match those specified
in the datafile used for Thermochimica calculations. Similarly, if `output_species` is a non-empty list
of `phase:species` pairs, these will be parsed and the specified species in the specified phases will be

> located in Thermochimica output and stored for later output via an AuxKernel.

Reviewer 2: Do we need these guidelines anymore since `ChemicalCompositionAction` sets up the UO and it's parameters are usually already checked there?

Author: removed

## Description

[`ThermochimicaElementData`](ThermochimicaElementData.md) performs Thermochimica calculations on elements.

!alert note

> This object can only be set up using teh [ChemicaComposition](ChemicaCompositionAction.md) action

Reviewer 2: ```suggestion
This object can only be set up using the [ChemicaComposition](ChemicaCompositionAction.md) action.
```
